src/EFCore.Relational/Query/SqlExpressions/UpdateExpression.cs

```diff
@@ -0,0 +1,126 @@
+    /// </summary>
+    /// <param name="table">A table on which the update operation is being applied.</param>
+    /// <param name="selectExpression">A select expression which is used to determine which rows to update and to get data from additional tables.</param>
+    /// <param name="setColumnValues">A list of <see cref="SetColumnValue"/> which specifies columns and their corresponding values to update.</param>
+    public UpdateExpression(TableExpression table, SelectExpression selectExpression, IReadOnlyList<SetColumnValue> setColumnValues)
+    {
+        Table = table;
+        SelectExpression = selectExpression;
+        SetColumnValues = setColumnValues;
+    }
+
+    /// <summary>
+    ///     The table on which the update operation is being applied.
+    /// </summary>
+    public TableExpression Table { get; }
+
+    /// <summary>
+    ///     The select expression which is used to determine which rows to update and to get data from additional tables.
+    /// </summary>
+    public SelectExpression SelectExpression { get; }
+
+    /// <summary>
+    ///     The list of <see cref="SetColumnValue"/> which specifies columns and their corresponding values to update.
+    /// </summary>
+    public IReadOnlyList<SetColumnValue> SetColumnValues { get; }
+
+    /// <inheritdoc />
+    public override Type Type
+        => typeof(object);
+
+    /// <inheritdoc />
+    public sealed override ExpressionType NodeType
+        => ExpressionType.Extension;
+
+    /// <inheritdoc />
+    protected override Expression VisitChildren(ExpressionVisitor visitor)
+    {
+        var selectExpression = (SelectExpression)visitor.Visit(SelectExpression);
+        var setValueExpressions = SetColumnValues.Select(e => new SetColumnValue(e.Column, (SqlExpression)visitor.Visit(e.Value))).ToList();
+
+        return Update(selectExpression, setValueExpressions);
+    }
+
+    /// <summary>
+    ///     Creates a new expression that is like this one, but using the supplied children. If all of the children are the same, it will
+    ///     return this expression.
+    /// </summary>
+    /// <param name="selectExpression">The <see cref="SelectExpression" /> property of the result.</param>
+    /// <param name="setValueExpressions">The <see cref="SetColumnValues" /> property of the result.</param>
+    /// <returns>This expression if no children changed, or an expression with the updated children.</returns>
+    public UpdateExpression Update(SelectExpression selectExpression, IReadOnlyList<SetColumnValue> setValueExpressions)
+        => selectExpression != SelectExpression || !SetColumnValues.SequenceEqual(setValueExpressions)
+            ? new UpdateExpression(Table, selectExpression, setValueExpressions)
+            : this;
+
+    /// <inheritdoc />
+    public void Print(ExpressionPrinter expressionPrinter)
+    {
+        expressionPrinter.AppendLine($"Update {Table.Name} AS {Table.Alias}");
```

Comment:
Don't we usually approximate the SQL?

```suggestion
        expressionPrinter.AppendLine($"UPDATE {Table.Name} AS {Table.Alias}");
```